### DISCUSSION TITLE
Egg file generated by easy-install works for multiple python versions

### DISCUSSION THREAD

**Discussion by chesskingaka** [pypa/setuptools#4114] [Q&A] [answered]
Hi,
I am asking you if we have any way to make egg file work for multiple python versions. 
I installed the package with setuptools to my PYTHONPATH with python3.9.5. I saw the egg file named mypack-1.0.0-**py3.9**.egg was created.

if I used the python 3.9.5 to run the script that includes: `__requires__='mypack=1.0.0'`, it works fine. But when i used the python 3.10.4 or python 3.11.5 to run the script, the error "DistributionNotFound." is returned

 if I change the name of the egg file to "mypack-1.0.0-**py3.10**.egg" or "mypack-1.0.0-**py3.11**.egg" then it works when i run that script with python3.10 or python3.11

**Comment by jwodder** [comment, marked as answer]
Why are you working with egg files?  They've been deprecated for years.  You should now be using wheels, which *do* work with multiple Python versions (assuming pure Python with no compiled extensions).

**Comment by abravalheri** [comment]
As pointed by @jwodder `eggs` were deprecated long ago, and more importantly we no longer support them.

So I guess the answer for your question would be "No".

**Comment by chesskingaka** [comment]
Thank you, guys, Actually, i am using the paf repo https://github.com/Ericsson/paf for my project, so  i followed the installation of README file and facing the issue as above.

I will try with the way you suggest!
